src/peer_manager.rs

```diff
@@ -652,17 +636,18 @@ impl PeerManager {
     /// Removes the given peer. Returns whether the peer was actually present.
     /// If the peer was joining before, it is demoted back to JoiningNode and false is returned.
     pub fn remove_peer(&mut self, pub_id: &PublicId) -> bool {
-        if let Some(mut peer) = self.peers.remove(pub_id) {
-            if peer.is_or_was_joining_node() && peer.is_node() {
-                peer.state = PeerState::JoiningNode;
-                let _ = self.peers.insert(peer.pub_id, peer);
-                return false;
+        match self.peers.entry(*pub_id) {
```

Comment:
Why the refactor here? not clear the code looks clearer.

Reply:
I prefer to use the [entry API](https://doc.rust-lang.org/std/collections/btree_map/enum.Entry.html) in cases where the alternative is to use double lookup (`contains_key` followed by `insert` or `remove` followed by `insert`, etc...). It is more efficient and I think also cleaner.